### ISSUE TITLE
changing the camera upvector lets visual objects disappear

### ISSUE THREAD

**Issue by rainergericke** [vsg-dev/vsgExamples#142]
Hi Robert,

for project::chrono we need an optional upvector (0,1,0) for backward compatibility to older model files. Changing the upvector in vsg programs let all visual objects vanish yet. vsgskybox is a good example for testing this. I guess it is not intended.

**Comment by robertosfield** [comment]
On Fri, 18 Mar 2022 at 10:20, Rainer Gericke ***@***.***>
wrote:

> Hi Robert,
>
> for project::chrono we need an optional upvector (0,1,0) for backward
> compatibility to older model files. Changing the upvector in vsg programs
> let all visual objects vanish yet. vsgskybox is a good example for testing
> this. I guess it is not intended.
>
> Could you provide a modified example to illustrate the problem and the
settings you are using.

Is it possible to share the problem data?

**Comment by rainergericke** [comment]
Yes here is it. It is just the standard vsgskybox example. With upvector (0,0,1) it works, building with upvector (0,1,0) it shows nothing but the background.

Both versions have been started with (on MacOS 12.3)

upvector models/teapot.vsgt --skybox textures/skybox.dds

[Upvector.zip](https://github.com/vsg-dev/vsgExamples/files/8304296/Upvector.zip)

**Comment by rainergericke** [comment]
Oh, maybe I know, what happend. The line of sight is parallel to y, when y is used as an upvector, a singularity occurs.

**Comment by robertosfield** [comment]
Good to hear you've found a way to resolve the issue.

It might be that the vsg::LookAt code needs a review to see if can do a
better job at handling the look vector and up vector being aligned.  I
presume this is the root of the problem from your description.  There is
only so much one can do though with invalid combinations of inputs.
Message ID: ***@***.***
com>
>
>
